### PR TITLE
[#140080] enable product search without logging in or when acting as

### PR DESCRIPTION
**Pull request by meara**
https://pm.tablexi.com/issues/140080

### CODE REVIEW

File: app/controllers/global_search_controller.rb [hunk 1]
```diff
@@ -14,9 +12,14 @@ def self.searcher_classes
   end
 
   def index
-    @searchers = self.class.searcher_classes.map do |searcher_class|
-      searcher_class.new(current_user, current_facility, params[:search])
-    end
+    @searchers = if session_user.nil? || session_user != acting_user
```
Comment (jhanggi) [inline app/controllers/global_search_controller.rb]: Looks like we have a `acting_as?` on ApplicationController that does something similar. Could we use that? I think Devise gives us a `logged_in?` helper. Would `if logged_in? && !acting_as?` be more intention-revealing? (I think what I said requires would flip the conditional)

File: app/controllers/global_search_controller.rb [hunk 2]
```diff
@@ -14,9 +12,13 @@ def self.searcher_classes
   end
 
   def index
-    @searchers = self.class.searcher_classes.map do |searcher_class|
-      searcher_class.new(current_user, current_facility, params[:search])
-    end
+    @searchers = if user_signed_in? && !acting_as?
```
Comment (jhanggi) [inline app/controllers/global_search_controller.rb]: I am somewhat of the opinion that you should be able to do a little more than you currently can while acting as another user. If we included other searchers, like the order detail or statement searchers, would that lead to some 403s or "you can't do that while acting as another user" from the resulting links? If it does, then this is fine.

If it doesn't I would vote towards including the searchers if you're signed in and using the user you're acting as as the permission check (I forget who is acting_user and who is current_user in this context).

File: spec/controllers/global_search_controller_spec.rb [hunk 3]
```diff
@@ -40,15 +40,23 @@ def it_should_have_customer_paths
...
   let(:statement_results) { assigns[:searchers].find { |s| s.template == "statements" }.results }
+  let(:product_results) { assigns[:searchers].find { |s| s.template == "products" }.results }
 
   describe "index" do
-    describe "permissions" do
-      before :each do
-        @action = :index
-        @method = :get
+
+    context "when not signed in" do
```
Comment (jhanggi) [inline spec/controllers/global_search_controller_spec.rb]: We should probably add a spec for when you're acting as another user.

File: app/controllers/global_search_controller.rb (hunk 2 above)
Comment (meara) [reply app/controllers/global_search_controller.rb]: You'll get a `This function is unavailable while you are acting as another user.` message if you try to click on an order detail link after searching for it.

**Pull request merged**